### PR TITLE
Fix #4064: Introduce admin controls UI for learner analytics monitoring in user studies

### LINKED ISSUE #4064: Add basic learner analytics support
PRD: https://docs.google.com/document/d/13H_r47Ge6dj_VAszEHQttoyOoCKXeEgodSTv16-7fnY/edit.

Between #4247 and #4248, remaining work items I've identified:
- [x] Disable the feature by default
- [x] Verify that event uploading is actually happening, and correctly, to Firebase
- [x] Shift the implementation such that the domain changes update modules, but don't actually change the application classes until the integration PR (revised to 4 PRs: domain components, integration, events, UI)
- [x] Generating device ID using a GUID that's persisted on disk (this is a **requirement change** since we've discovered that we can't rely on hardware identifiers)
- [x] Update event logging flow to be asynchronous since the device ID has to be persisted on disk and thus can't be retrieved synchronously (see previous item) -- Update: only the IDs part needs to be asynchronous (which conveniently builds nicely into existing controllers now that using ``StateFlow`` is on the table since I branched the PRs off of math expressions).
- [ ] Finish implementing the remaining events, and make sure they're thoroughly tested
- [x] Session ID needs to be piped using data providers (which might introduce a situation where we actually block on #4239 which would not be ideal)
- [ ] All tests need to be checked to pass on Gradle & Bazel
- [ ] Affected UI tests need to be tested with Espresso via Gradle
- [x] We may potentially need the Bazel test fixes introduced in #4068 since these PRs are affecting so many tests it's causing the compute affected tests workflow to fail
- [x] Any other comments I've listed in #4247 and #4248
- [x] Figure out the chaining & merge strategy for the project since changes are sort of messily split between the two PRs (maybe partly because of earlier merging challenges). It'd be nice to keep PR reviews as simple as possible, but the priority needs to be execution speed.
- [x] Since new branches are going to be created per the merging strategy, ensure all the changes from the early project are fully covered within those branches

Statuses on all of the events that need to be added:
- [x] Card Start: not started as of the opening of this issue
- [x] Card End: not started as of the opening of this issue
- [x] Hint Offered: not started as of the opening of this issue
- [x] Hint Accessed: not started as of the opening of this issue
- [x] Solution accessed: not started as of the opening of this issue
- [x] Solution offered: not started as of the opening of this issue
- [x] Submit answer: not started as of the opening of this issue
- [x] Play voice over: not started as of the opening of this issue
- [x] Background app inactivity: not started as of the opening of this issue
- [x] Foreground app: not started as of the opening of this issue
- [x] Exit exploration: not started as of the opening of this issue
- [x] Finish lesson: started to be logged as of the opening of this issue, but needs to be fixed & tested
- [x] Resume lesson: not started as of the opening of this issue
- [x] Click “Start Over” on resume: not started as of the opening of this issue
- [x] Delete profile: not started as of the opening of this issue

Items listed in the TDD that I'm not implementing to save time (these are being dropped as requirements)
- Implementing last synced time in the UI
- Implementing a snackbar for when text is copied in the admin panel (this doesn't seem necessary anymore since the button transitions to a 'copied' state; it seems super clear)

### PR DESCRIPTION
**Pull request by BenHenning**
## Explanation

Fixes #4064
This PR is a rebase of #4262 to remove #2173 & its predecessors.

### Key Changes:
Introduced a new admin menu option: **ProfileLearnerIds and SyncStatus** via the `ProfileAndDeviceIdActivity`. This allows admins to view the learner analytics screen, and do the following:
- Copy the installation ID to clipboard
- Copy the Profile Name and ProfileId to clipboard
- View the sync status of events
- Peform a manual sync, if there are events awaiting upload
- Export and share a copy of all IDs and Logs

### Other Changes
1. This PR removed the `AnalyticsController.logTransitionEvent` function and replaced its usages with the `AnalyticsController.logImportantEvent` function.
2. `StoryFragmentPresenter.kt`, `ResumeLessonFragmentPresenter.kt` and `RecentlyPlayedFragmentPresenter.kt` have been refactored to better manage start playing, resume playing, and restart an exploration.
3. `com.google.android.material:material`  was updated to  `1.3.0`.

## Essential Checklist

- [ ] The PR title and explanation each start with "Fix #bugnum: " (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [ ] Any changes to [scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets) files have their rationale included in the PR explanation.
- [ ] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [ ] The PR does not contain any unnecessary code changes from Android Studio ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
- [ ] The PR is made from a branch that's **not** called "develop" and is up-to-date with "develop".
- [ ] The PR is **assigned** to the appropriate reviewers ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).

### Phone
|||
|--|--|
|Screenshot_2025-04-17-22-21-33-703_org oppia android|Screenshot_2025-04-17-22-21-47-543_org oppia android|
|Screenshot_2025-04-17-22-25-09-087_org oppia android|Screenshot_2025-04-17-22-23-53-347_org oppia android|

In the video linked below, there is a bug where I had to go back to the previous screen before the analytics activity could detect network change, and enable the button to manually upload the logs.

https://drive.google.com/file/d/129SV-_DWxGjxInKVgwCrErINyNKzGbkX/view?usp=sharing

### CODE REVIEW

**Comment by BenHenning** [comment]
Still active.

**Comment by BenHenning** [comment]
@anandwana001 PTAL. This is the only PR that's not 100% done yet, but it's quite close (everything except a few test cases has been implemented, and there are some bugs that were found in the new UI that I'll likely defer to follow-up work items or outright turn down since this is a study-only UI). The CI failures are due to just 1 test being broken, and that should be straightforward to fix.

PTAL as if the PR is ready, and feel free to block approval if you're not comfortable approving with some pieces missing. If you are comfortable approving, I'll ask for a follow-up double check of the changes that come post-approval.

**Comment by BenHenning** [comment]
Note that any additional changes needed in this PR will be covered in #4345 as a fast-follow.

Furthermore, @anandwana001's and @rt4914's reviews are being deferred (#4344 and #4266 are tracking each, respectively). Note that this does result in no approvals, but I've gotten confirmation from @seanlip as another technical representative that this is reasonable given the circumstances of getting these changes in quickly for an upcoming release. I'll make sure this PR does get reviewed post-submission.

**Review by anandwana001 — COMMENTED** [review]
LGTM

**Review by rt4914 — COMMENTED** [review]
LGTM. Thanks.

File: app/src/main/java/org/oppia/android/app/administratorcontrols/AdministratorControlsActivityPresenter.kt
Comment (adhiamboperes) [inline app/src/main/java/org/oppia/android/app/administratorcontrols/AdministratorControlsActivityPresenter.kt:1]: UI tests need to be added for these changes:
- verify that ProfileAndDeviceId option is displayed in multiplane layouts
- verify that ProfileAndDeviceId option is displayed in non-multiplane layouts

File: app/src/main/java/org/oppia/android/app/player/audio/AudioFragment.kt
Comment (adhiamboperes) [inline app/src/main/java/org/oppia/android/app/player/audio/AudioFragment.kt:1]: This file is currently exempted from tests. We need to add tests for it.

File: app/src/main/java/org/oppia/android/app/profile/ProfileChooserFragmentPresenter.kt [hunk 1]
```diff
@@ -251,8 +251,13 @@ class ProfileChooserFragmentPresenter @Inject constructor(
   }
 
   private fun logProfileChooserEvent() {
-    oppiaLogger.logTransitionEvent(
-      oppiaClock.getCurrentTimeMs(), eventContext = oppiaLogger.createOpenProfileChooserContext()
-    )
+    oppiaLogger.logImportantEvent(oppiaLogger.createOpenProfileChooserContext())
```
Comment (adhiamboperes) [inline app/src/main/java/org/oppia/android/app/profile/ProfileChooserFragmentPresenter.kt:254]: We need tests for this

File: app/src/main/java/org/oppia/android/app/resumelesson/ResumeLessonFragmentPresenter.kt
Comment (adhiamboperes) [inline app/src/main/java/org/oppia/android/app/resumelesson/ResumeLessonFragmentPresenter.kt:1]: Tests are needed for:
1. checkpoint retrieved successfully
2. checkpoint retrieval failed
3. click restart lesson
4. click continue lesson.

File: app/src/main/java/org/oppia/android/app/story/StoryFragmentPresenter.kt
Comment (adhiamboperes) [inline app/src/main/java/org/oppia/android/app/story/StoryFragmentPresenter.kt:1]: Tests are needed for:

1. Resuming an in progress chapter(navigate to resume lesson activity)
2. Replaying a completed chapter(navigate to exploration activity)
3. Playing a chapter not started yet(navigate to exploration activity)
4. EventLog(createOpenStoryActivityContext)

File: app/src/main/java/org/oppia/android/app/topic/conceptcard/ConceptCardFragmentPresenter.kt [hunk 2]
```diff
@@ -85,9 +83,6 @@ class ConceptCardFragmentPresenter @Inject constructor(
   }
 
   private fun logConceptCardEvent(skillId: String) {
-    oppiaLogger.logTransitionEvent(
-      oppiaClock.getCurrentTimeMs(),
-      oppiaLogger.createOpenConceptCardContext(skillId)
-    )
+    oppiaLogger.logImportantEvent(oppiaLogger.createOpenConceptCardContext(skillId))
```
Comment (adhiamboperes) [inline app/src/main/java/org/oppia/android/app/topic/conceptcard/ConceptCardFragmentPresenter.kt:86]: Add tests for eventlog

File: app/src/main/res/layout/administrator_controls_learner_analytics_view.xml
Comment (adhiamboperes) [inline app/src/main/res/layout/administrator_controls_learner_analytics_view.xml:1]: This and the corresponding alternate layouts need tests to verify the contents.

**Review by adhiamboperes — COMMENTED** [review]
Noted all the test gaps in the PR. These are tracked in https://github.com/oppia/oppia-android/issues/5815

**Comment by seanlip** [comment]
@adhiamboperes Just checking -- this is a closed PR. Does some follow-up issue need to be filed for the test gaps?

If so please link to it in your top-level comment, thanks!

**Comment by adhiamboperes** [comment]
@seanlip, yes, I reviewed this as part of https://github.com/oppia/oppia-android/issues/4345. I will be filing an issue to track these test gaps.

**Pull request merged**